GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -0,0 +1,185 @@
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqualObjects(authFlow.error, expectedError);
+}
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+- (void)testGenerateAuthFlowWithMissingAuthorizationResponse {
+  NSError *error = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                       code:GIDVerifyErrorCodeUnknown
+                                   userInfo:nil];
+
+  NSString *errorString = [error localizedDescription];
+  NSDictionary<NSString *, NSString *> *errorDict = @{ NSLocalizedDescriptionKey : errorString };
+  NSError *expectedError = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                               code:GIDVerifyErrorCodeUnknown
+                                           userInfo:errorDict];
+
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerifyAccountDetail
+                                                               configuration:nil
+                                                                       error:error];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqualObjects(authFlow.error, expectedError);
+}
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+- (void)testMaybeFetchToken_authFlowError {
+  NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
+                                       code:kGIDSignInErrorCodeUnknown
+                                   userInfo:nil];
+
+  OIDTokenResponse *tokenResponse = 
+      [OIDTokenResponse testInstanceWithAccessTokenExpiration:@(kAccessTokenExpiration)];
+  OIDAuthState *authState = [OIDAuthState testInstanceWithTokenResponse:tokenResponse];
+
+  GIDAuthorizationResponseHandlingFake *responseHandler = 
+      [[GIDAuthorizationResponseHandlingFake alloc] initWithAuthState:authState error:nil];
+  GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] initWithAuthState:nil
+                                                           error:error
+                                                      emmSupport:nil
+                                                     profileData:nil];
+
+  [responseHandler maybeFetchToken:authFlow];
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqualObjects(authFlow.error, error);
+}
+
+- (void)testMaybeFetchToken_noRefresh {
+  NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
```

Comment:
FWIW I like naming things like `error` here as `expectedError` so it's clear that this is the error we are making sure we have.

Reply:
Sounds good, applied.